### ISSUE TITLE
When padding is big int, torch.nn.functional.fold runs too long and can't return result

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83152]
### 🐛 Describe the bug

When I run the code, there is no error information reports. After 5 mins running, there is no response and I can't stop the cmd, I have to kill the cmd.

```
import torch
results={}
arg_1_tensor = torch.rand([1, 12, 12], dtype=torch.float32)
arg_1 = arg_1_tensor.clone()
arg_2 = [4,5,]
arg_3 = [2,2,]
arg_4 = 1
arg_5 = 36028797018963968
arg_6 = 1
results['res'] = torch.nn.functional.fold(arg_1,arg_2,arg_3,arg_4,arg_5,arg_6,)
```

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11
